### PR TITLE
docs - add GUC pgcrypto.fips

### PR DESCRIPTION
**Pull request by mkiyama**
also, see related doc PR for pgcrypto extension https://github.com/pivotal/gp-docs-pivotal-gpdb/pull/172

dev PR for pgcrypto 5X_STABLE https://github.com/greenplum-db/gpdb/pull/8542

This is 5X_STABLE only, for now

Updates in HTML format
pgcrypto.fips guc
http://docs-msk-gpdb5-dev.cfapps.io/5220/ref_guide/config_params/guc-list.html#pgcrypto.fips

pgcrypto extension
http://docs-msk-gpdb5-dev.cfapps.io/5220/install_guide/install_pgcrypto.html

### CODE REVIEW

File: gpdb-doc/dita/ref_guide/config_params/guc-list.xml [hunk 1]
```diff
@@ -8486,6 +8488,65 @@
...
+          <li>RAW encryption and decryption functions support only AES and 3DES (other algorithms
+            such as blowfish are not allowed)</li>
+        </ul></p>
+      <p>To enable <codeph>pgcrypto.fips</codeph>:<ol id="ol_hrj_vhq_gjb">
+          <li>Enable the <codeph>pgcrypto</codeph> functions as an extension. See <xref
+              href="../../install_guide/install_pgcrypto.xml" scope="peer">pgcrypto Cryptographic
+              Functions</xref>.</li>
+          <li>Set the <codeph>pgcrypto.fips</codeph> server configuration parameter to
+              <codeph>on</codeph> for each database that uses FIPS encryption. For example, this
+            command sets the parameter to on for the database <codeph>testdb</codeph>. <codeblock># ALTER DATABASE testdb SET pgcrypto.fips TO on;</codeblock>
```
Comment (litzell) [inline gpdb-doc/dita/ref_guide/config_params/guc-list.xml]: sets the parameter to <codeph>on</codeph>?

**Comment by mkiyama** [comment]
Added the steps for installing/uninstalling the $libdir/pgcrypto library from the pgcrypto topic to keep all the FIPS information in one place.
see PR https://github.com/pivotal/gp-docs-pivotal-gpdb/pull/172

I also updated the HTML output.

File: gpdb-doc/dita/ref_guide/config_params/guc-list.xml [hunk 2]
```diff
@@ -8486,6 +8488,94 @@
...
+          <li>The functions for crypt and gen_salt algorithms are disabled</li>
+          <li>PGP encryption and decryption functions support only AES and 3DES encryption
+            algorithms (other algorithms such as blowfish are not allowed)</li>
+          <li>RAW encryption and decryption functions support only AES and 3DES (other algorithms
+            such as blowfish are not allowed)</li>
+        </ul></p>
+      <p><b>To enable <codeph>pgcrypto.fips</codeph></b><ol id="ol_hrj_vhq_gjb">
+          <li>Enable the <codeph>pgcrypto</codeph> functions as an extension if it is not enabled.
+            See <xref href="../../install_guide/install_pgcrypto.xml" scope="peer">pgcrypto
+              Cryptographic Functions</xref>. This example <codeph>psql</codeph> command drops the
```
Comment (dgkimura) [inline gpdb-doc/dita/ref_guide/config_params/guc-list.xml]: I think s/drops/creates/

File: gpdb-doc/dita/ref_guide/config_params/guc-list.xml [hunk 3]
```diff
@@ -8486,6 +8488,94 @@
...
+          <xref
+          href="https://www.nist.gov/itl/popular-links/federal-information-processing-standards-fips"
+          format="html" scope="external"
+          >https://www.nist.gov/itl/popular-links/federal-information-processing-standards-fips</xref>.
+        The default is <codeph>off</codeph>, encryption is not enabled. </p>
+      <p>When this parameter is enabled, these changes occur:<ul id="ul_qn1_jwh_gjb">
+          <li>FIPS mode is initialized in the OpenSSL library</li>
+          <li>The functions <cmdname>digest()</cmdname> and <cmdname>hmac()</cmdname> allow only the
+            SHA encryption algorithm (MD5 is not allowed)</li>
+          <li>The functions for crypt and gen_salt algorithms are disabled</li>
```
Comment (dyozie) [inline gpdb-doc/dita/ref_guide/config_params/guc-list.xml]: for crypt -> for the crypt

**Comment by mkiyama** [comment]
@dgkimura 
Do you know if FIPS needs to enabled on the host systems?
Thanks.

**Comment by dgkimura** [comment]
On Mon, Oct 7, 2019 at 8:43 AM Mel Kiyama wrote:
> Do you know if FIPS needs to enabled on the host systems?

Yes, if the customer is using fips mode then it should be.  If they are not, then it doesn't need to be.

**Pull request merged**